### ISSUE TITLE
[BUG] torch2trt assert(permutation[0] == 0)  # cannot move batch dim

### ISSUE THREAD

**Issue by Ken1256** [arogozhnikov/einops#207]
**Describe the bug**
https://github.com/cszn/SCUNet/blob/main/models/network_scunet.py

```
  File "C:\Program Files\Python310\lib\site-packages\einops\einops.py", line 487, in rearrange
    return reduce(tensor, pattern, reduction='rearrange', **axes_lengths)
  File "C:\Program Files\Python310\lib\site-packages\einops\einops.py", line 410, in reduce
    return _apply_recipe(recipe, tensor, reduction_type=reduction)
  File "C:\Program Files\Python310\lib\site-packages\einops\einops.py", line 236, in _apply_recipe
    tensor = backend.transpose(tensor, axes_reordering)
  File "C:\Program Files\Python310\lib\site-packages\einops\_backends.py", line 331, in transpose
    return x.permute(axes)
  File "C:\Program Files\Python310\lib\site-packages\torch2trt-0.4.0-py3.10.egg\torch2trt\torch2trt.py", line 307, in wrapper
    converter["converter"](ctx)
  File "C:\Program Files\Python310\lib\site-packages\torch2trt-0.4.0-py3.10.egg\torch2trt\converters\permute.py", line 17, in convert_permute
    assert(permutation[0] == 0)  # cannot move batch dim
```

**Reproduction steps**
```
if __name__ == '__main__':

    from torch2trt import torch2trt
    
    device = 'cuda'
    net = SCUNet().eval().to(device)
    x = torch.randn((1, 3, 128, 128)).to(device)
    
    net = torch2trt(net, [x])
    
    x = net(x)
    print(x.shape)
```

**Expected behavior**
Maybe:
https://github.com/NVIDIA-AI-IOT/torch2trt/issues/742#issuecomment-1170672478

**Your platform**
```
Win10 x64

Package                 Version
----------------------- --------------------
absl-py                 1.2.0
addict                  2.4.0
basicsr                 1.4.2
cachetools              5.2.0
certifi                 2022.9.14
charset-normalizer      2.1.1
colorama                0.4.5
contourpy               1.0.5
cycler                  0.11.0
einops                  0.4.1
fairscale               0.4.9
focal-frequency-loss    0.3.0
fonttools               4.37.3
future                  0.18.2
google-auth             2.11.1
google-auth-oauthlib    0.4.6
graphsurgeon            0.4.6
grpcio                  1.49.1
idna                    3.4
imageio                 2.22.0
kiwisolver              1.4.4
lmdb                    1.3.0
lpips                   0.1.4
Markdown                3.4.1
MarkupSafe              2.1.1
matplotlib              3.6.0
networkx                2.8.6
numpy                   1.23.3
oauthlib                3.2.1
onnx                    1.12.0
onnx-graphsurgeon       0.3.12
opencv-python           4.6.0.66
packaging               21.3
Pillow                  9.2.0
pip                     22.2.2
protobuf                3.20.1
pyasn1                  0.4.8
pyasn1-modules          0.2.8
pyparsing               3.0.9
python-dateutil         2.8.2
PyWavelets              1.4.1
PyYAML                  6.0
requests                2.28.1
requests-oauthlib       1.3.1
rsa                     4.9
scikit-image            0.19.3
scipy                   1.9.1
setuptools              63.2.0
six                     1.16.0
tb-nightly              2.11.0a20220922
tensorboard-data-server 0.6.1
tensorboard-plugin-wit  1.8.1
tensorrt                8.4.3.1
thop                    0.1.1.post2209072238
tifffile                2022.8.12
timm                    0.6.7
torch                   1.12.1+cu116
torch2trt               0.4.0
torchaudio              0.12.1+cu116
torchinfo               1.7.0
torchvision             0.13.1+cu116
tqdm                    4.64.1
typing_extensions       4.3.0
uff                     0.6.9
urllib3                 1.26.12
Werkzeug                2.2.2
wheel                   0.37.1
yapf                    0.32.0
```

**Comment by arogozhnikov** [comment]
Hi @Ken1256,
This error is a result of using permutation that is not supported by trt, you need to change your model.

It is not an einops issue, it is about your computation graph being not supported by trt
